controllers/platformnetwork_controller.go

```diff
@@ -146,7 +148,7 @@ func oamUpdateRequired(instance *starlingxv1.PlatformNetwork, p *oamNetworks.OAM
 // include in the request options to minimum churn and to ease debugging.
 func poolUpdateRequired(instance *starlingxv1.PlatformNetwork, p *addresspools.AddressPool, r *PlatformNetworkReconciler) (opts addresspools.AddressPoolOpts, result bool) {
 	var delta strings.Builder
-	if instance.Name != p.Name {
+	if r.GetAddrPoolNameByNetworkType(instance.Spec.Type, instance.Name) != p.Name {
```

Comment:
> Can we add a comment here referring to the rule about the mgmt address pool name? We need to keep tracking this since it's possible to change in the future.

Done.